ui/apps/platform/src/Components/NotifierConfiguration/NotifierConfigurationForm.tsx

```diff
@@ -0,0 +1,241 @@
+import React, { ReactElement, useState } from 'react';
+import { Button, Card, CardBody, CardTitle, Flex, FlexItem, Tooltip } from '@patternfly/react-core';
+import { HelpIcon, PencilAltIcon, PlusCircleIcon, TrashIcon } from '@patternfly/react-icons';
+import { FormikErrors } from 'formik';
+
+import { isDefaultEmailTemplate } from 'Components/EmailTemplate/EmailTemplate.utils';
+import EmailTemplateModal, {
+    TemplatePreviewArgs,
+} from 'Components/EmailTemplate/EmailTemplateModal';
+import FormLabelGroup from 'Components/PatternFly/FormLabelGroup';
+import useIndexKey from 'hooks/useIndexKey';
+import { NotifierIntegrationBase } from 'services/NotifierIntegrationsService';
+import { NotifierConfiguration } from 'services/ReportsService.types';
+
+import NotifierMailingLists from './NotifierMailingLists';
+
+function splitAndTrimMailingListsString(mailingListsString: string): string[] {
+    return mailingListsString.split(',').map((email) => email.trim());
+}
+
+export type NotifierConfigurationFormProps = {
+    customBodyDefault: string;
+    customSubjectDefault: string;
+    errors: FormikErrors<unknown>;
+    fieldIdPrefix: string;
+    hasWriteAccessForIntegration: boolean;
+    notifierConfigurations: NotifierConfiguration[];
+    onDeleteLastNotifierConfiguration?: () => void;
+    renderTemplatePreview?: (args: TemplatePreviewArgs) => ReactElement;
+    setFieldValue: (fieldId: string, value: unknown) => void;
+};
+
+function NotifierConfigurationForm({
+    customBodyDefault,
+    customSubjectDefault,
+    errors,
+    fieldIdPrefix,
```

Comment:
Not a change request but I did have to stare at this for a bit to figure out what it's accomplishing. I think a different name might have clued me in quicker, something like `formikValueKey` or `formikFieldKey`. With this prop being so tightly coupled with formik I don't think it'd be bad to include it in the prop name.

Reply:
Is `fieldIdPrefixFormikPatternFly` too verbose?
* Formik: first argument of `setFieldValue` function.
* PatternFly: `fieldId` property.

Reply:
@pedrottimark what are the possible values for `fieldIdPrefix`? Maybe that would help us understand what kind of name would be more appropriate. From the comment below it seems like we could have `deliveryDestinations` or `notifierConfigurations`. Is that correct or is it supposed to be more generic?

Reply:
Yeah, I will add comments with examples in every component:

In `NotifierConfigurationForm` component
* Vulnerability Reports: `deliveryDestinations`
* Compliance Reports might have an extra level: `report.notifierConfigurations`

Within `map` method to render `NotifierMailingLists` element:
* Vulnerability Reports: `deliveryDestinations[0]` and so on
* Compliance Reports might have an extra level: `eport.notifierConfigurations[0]` and so on